app/services/rapid_ready_for_decision/hypertension_pdf_generator.rb

```diff
@@ -8,17 +8,29 @@ def initialize(patient, blood_pressure_data, medications)
       @blood_pressure_data = blood_pressure_data
       @medications = medications
       @date = Time.zone.today
+
+      @pdf.markup_options = {
+        text: {
+          size: 11
+        },
+        heading4: {
+          margin_top: 12
+        },
+        table: {
+          width: 150,
+          cell: {
+            size: 10,
+            border_width: 0,
+            background_color: 'f3f3f3'
+          }
+        }
+      }
     end
 
-    # progressively builds a pdf and is sensitive to sequence
     def generate
-      add_intro
-      add_blood_pressure_intro
-      add_blood_pressure_list
-      add_blood_pressure_outro
-      add_medications_intro
-      add_medications_list
-      add_about
+      template = File.read('app/services/rapid_ready_for_decision/views/hypertension_pdf.erb')
+
+      @pdf.markup ERB.new(template).result(binding)
```

Comment:
What Rails magic 🪄 is this? 
* `ERB.new(template)` loads the file as an ERB object
* What does `result(binding)` do?

Reply:
> What Rails magic 🪄 is this?
> 
> * `ERB.new(template)` loads the file as an ERB object
> * What does `result(binding)` do? I presume it binds the instance variables (e.g., `@blood_pressure_data`) to the context of the new ERB object. Is this how the `prawn-markup` gem should be used?

Yeah, I was surprised, too. This actually takes a cue from how mailer templates are handled ([for example](https://github.com/department-of-veterans-affairs/vets-api/blob/53db8ed5d7a8b2eab2552f1edd7081ade7ae5d36/app/mailers/hypertension_fast_track_pilot_mailer.rb#L19)). That's mostly correct: it's not just instance variables, but the entire class context... @nanotone and I were looking through it the other day. Think of it as `this` in JavaScript... 

All `#result` does is generate a plain string that is passed to the #markup method. As for whether this is how `prawn-markup` _should_ be used, I would argue that it seems valid: we can think of ERB as a more advanced and feature-rich version of multiline strings. 

That said, I'm not married to this — there are definite trade-offs. I'd be happy to investigate another approach OR see if I can minimize those trade-offs even more.